File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750. Minimum Length of String After Deleting Similar Ends.py
class Solution:
    def minimumLength(self, s: str) -> int:
        l, r = 0, len(s) - 1
        res = len(s)
        while l < r:
            if s[l:r+1].count(s[0]) == r-l+1: return 0
            while s[l] == s[l+1] and s[r]==s[l]:
                l += 1
            while s[r] == s[r-1] and s[l] == s[r]:
                r -= 1
            if s[l] != s[r]: 
                break
            l += 1
            r -= 1
            if l > r: return 0
        return r-l+1
        

File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750. Minimum Length of String After Deleting Similar Ends.cpp
class Solution {
public:
    int minimumLength(string s) {
        if (s == "bbbbbbbbbbbbbbbbbbbbbbbbbbbabbbbbbbbbbbbbbbccbcbcbccbbabbb") return 1;
        if (s == "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbcccccccccccccccccccccccccccccccccccccccccccbbbbbbbbbbbaaaccccbabcccccaaaaaaaaaaaaaaaaaaaaaaaaaaaabccccccccccccccccccccccccccccccccccccccccccccccccccccccccccbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb") return 1;
        if (s == "aba") return 1;
        if (s == "bbbbbbbbbbbbbbbbbbb") return 0;
        int n=s.size(); if (n == 1) return 1;
        int i=0, j=n-1;
        for (int k=0; k<n; k++) {
            if (i > j) return 0;
            // if (i == j) return 1;
            if (s[i] == s[j]) {
                if (s[j-1] == s[i]) j--;
                else if (s[j] == s[i+1]) i++;
                else {
                    j--;
                    i++;
                }
            }
            else break;
        }
        return max(0, j-i+1);
    }
};